lib/orderbook/OrderBook.ts

```diff
@@ -96,18 +96,19 @@ class OrderBook extends EventEmitter {
     return this.addOwnOrder(order);
   }
 
-  public addMarketOrder = (_order: orders.OwnOrder) => {
-    // TODO: implement
+  public addMarketOrder = (order: orders.MarketOrder) => {
+    const price = order.quantity > 0 ? Number.MAX_VALUE : 0;
+    return this.addOwnOrder({ ...order, price }, true);
   }
 
-  private addOwnOrder = (order: orders.OwnOrder): matchingEngine.MatchingResult => {
+  private addOwnOrder = (order: orders.OwnOrder, discardRemaining: boolean = false): matchingEngine.MatchingResult => {
```

Comment:
if `discardRemaining` is `true`, don't broadcast or add to the orders list (line 120)